markets/spot-market/contracts/storage/AsyncOrder.sol

```diff
@@ -45,6 +49,13 @@ library AsyncOrder {
 
         token.transferFrom(from, address(this), synthAmount);
         asyncOrderData.totalEscrowedSynthShares += sharesAmount;
+
+        // sanity check to ensure the right shares amount is calculated
+        uint acceptableSynthAmount = convertSharesToSynth(asyncOrderData, marketId, sharesAmount) +
+            maxRoundingLoss;
```

Comment:
shouldn't we be subtracting maxRoundingLoss here?

Reply:
no, the `convertSharesToSynth` will never return _more_ synth than the synth amount sent in.  due to division, it loses some micro precision so adding the maxRoundingLoss to the conversion back